src/common/snippets/tests/src/pass/precision_propagation.cpp

```diff
@@ -0,0 +1,333 @@
+        std::ostringstream result;
+        result << "{";
+        for (const auto& precisions : precisions_pack) {
+            result << CommonTestUtils::vec2str(precisions) << "_";
+        }
+        result << "}";
+        return result.str();
+    };
+
+    std::ostringstream result;
+    result << "IN0_" << shapes.first << "_" << test_values.input_types[0] << "_"
+           << "IN1_" << shapes.second << "_" << test_values.input_types[1] << "_"
+           << "IN2_" << test_values.input_types[2]
+           << to_string(test_values.actual.op1_supported_precisions) << "_"
+           << to_string(test_values.actual.op2_supported_precisions) << "_"
+           << test_values.expected.convertion_before_op1.first << "_" << test_values.expected.convertion_before_op1.second << "_"
+           << test_values.expected.convertion_before_op2_1 << "_"
+           << test_values.expected.convertion_before_op2_2.first << "_" << test_values.expected.convertion_before_op2_2.second << "_"
+           << test_values.expected.convertion_after_op2 << "_";
+    return result.str();
+}
+
+TEST_P(PrecisionPropagationTest, CompareFunctions) {
+    disable_rt_info_check();
+
+    const auto param = GetParam();
+    const auto shapes = std::get<0>(param);
+    const auto test_values = std::get<1>(param);
+
+    const auto input_shapes = std::vector<PartialShape>({ shapes.first, shapes.second });
+    EmpyInfrastructureSupportStubForPrecisionPropagationFunction<DummyAdd> infrastructureSupportStub(
+        input_shapes,
+        test_values.input_types[0],
+        test_values.input_types[1],
+        test_values.input_types[2],
+        {
+            test_values.actual.convertion_before_op1,
+            test_values.actual.convertion_before_op2_1,
+            test_values.actual.convertion_before_op2_2
+        },
+        {
+            test_values.expected.convertion_before_op1,
+            test_values.expected.convertion_before_op2_1,
+            test_values.expected.convertion_before_op2_2,
+            test_values.expected.convertion_after_op2
+        });
+    function = infrastructureSupportStub.getOriginal();
+
+    const auto target_machine = std::make_shared<DummyPrecisionPropagationTargetMachine>(
+        test_values.actual.op1_supported_precisions,
+        test_values.actual.op2_supported_precisions);
+
+    manager.register_pass<ngraph::snippets::pass::PropagatePrecision>(target_machine);
+
+    function_ref = infrastructureSupportStub.getReference();
+}
+
+namespace PrecisionPropagationTestInstantiation {
+// clang-format off
+
+std::vector<std::pair<PartialShape, PartialShape>> shapes {
+    {{1, 3, 16, 16}, {1, 3, 16, 16}}
+};
+
+std::vector<PrecisionPropagationParamsValues> test_cases {
+    {
+        {element::f32, element::f32, element::f32},
+        {
+            {},
+            {},
+            {},
+            {{element::f32, element::f32}},
+            {{element::f32, element::f32}}
+        },
+        {}
```

Comment:
Forgot to add the last comment:
You pass `actual` and `expected` to every test case, however `expected` is uniquely defined by `actual`, right?
It looks like it's enough to pass only `actual` and then produce both initial and reference functions. This logic could be implemented inside `get_reference()` method, for example. What do you think?